modules/apps/foundation/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/servlet/FriendlyURLServlet.java

```diff
@@ -202,30 +203,34 @@ else if (_log.isWarnEnabled()) {
 		Map<String, String[]> params = request.getParameterMap();
 
 		try {
-			LayoutFriendlyURLComposite layoutFriendlyURLComposite =
-				portal.getLayoutFriendlyURLComposite(
-					group.getGroupId(), _private, friendlyURL, params,
-					requestContext);
+			LayoutFriendlyURLSeparatorComposite
+				layoutFriendlyURLSeparatorComposite =
```

Comment:
Hi @JorgeFerrer 

actually we did it. 

In the FriendlyURLResolver  there is a default method that return the new type LayoutFriendlyURLSeparatorComposite. This method wraps the old type and convert in the new type.

In this way we still support both.

If there is a better solution i will be happy to change the code.

Thanks